### DISCUSSION TITLE
Having issues importing private key in fastlane keychain

### DISCUSSION THREAD

**Discussion by tomvisions** [fastlane/fastlane#18379] [Q&A]
I have imported my private key, appstore certicate and appstore profile into match repo without issues.

Unfortunately, when match runs, it tries to import the private key into the fastlane keychain with no password. 

The private  Key I have does have a password. I've been trying to find out if there is a variable I can use to set the password, but have had no success.

Would anyone have any ideas?

**Comment by airtonix** [comment]
@tomvisions fastlane doesn't support certificates with passwords

You'll either need to re-export them and unpack them with the openssl command

```
openssl pkcs12 -in combined-cert-and-key.p12 -nocerts -out just-the-key.p12 -nodes
openssl pkcs12 -in combined-cert-and-key.p12 -nokeys -out just-the-cert.cert
```
or just let match create the cert and profiles. (my preference)

Apple resigns your app anyway when you pass their review process, and so the only thing you really need to worry about is that your application_id is controlled and owned by your apple connect developer account (or the team that you're a part of).

- https://stackoverflow.com/a/6320268/454615
